hooks/useCachedState.tsx

```diff
@@ -0,0 +1,17 @@
+import { useState, useEffect, Dispatch, SetStateAction } from "react";
+
+const stateMap = new Map();
+
+/* This hook is used to cache state between mount/umount of the same component 
+  It is meant to be used for things that are specific to one component only but have to be persisted between unmounts (table filters after a page transition for example). 
+*/
+export function useCachedState<T = unknown>(
+  key: string
+): [T, Dispatch<SetStateAction<T>>] {
+  const [state, setState] = useState<T>(stateMap.get(key));
+
+  useEffect(() => {
+    stateMap.set(key, state);
```

Comment:
good point, but moving the state up feels "unnatural" as this state would have to go in global state and is only relevant to the table, a cleaner approach would be to save the state to local storage, but reading from local storage can be expensive so I decided to just put it in memory. 
Regarding the updating, you are right and I think this could even be changed to be "useCachedReference" and this reference could be update imperatively and without re-rendering,  let me know if you prefer that approach and I can change it to match it

Reply:
To be honest, given a lot of data is fetched from the API, it being global state doesn't seem that unreasonable to me.

Reply:
i think this is basically global state, but without the typical global state patterns.  i cant really think of a better way to do it right now, but yea if you could come up with a way to just inline the updates to the map during the set state that probably would be an improvement. but its not that big of a deal so its up to you

Reply:
here is a possible update:
```javascript
export function useCachedState<T = unknown>(
  key: string
): [T, Dispatch<SetStateAction<T>>] {
  const [state, setState] = useState<T>(stateMap.get(key));

  const update = useCallback(
    (value: SetStateAction<T>) => {
      stateMap.set(key, value instanceof Function ? value(state) : value);
      setState(value);
    },
    [key, state]
  );
  return [state, update];
}
```

Do you feel like this solves the problem you've mentioned about the updating being indirect?

Reply:
> i think this is basically global state, but without the typical global state patterns. i cant really think of a better way to do it right now, but yea if you could come up with a way to just inline the updates to the map during the set state that probably would be an improvement. but its not that big of a deal so its up to you

To be honest, I think we just need to add some global state to this application on second thinking. Because the user can hop around a lot and we're waiting for their TX to go through, we're just resisting the reality of the app.

The data will get very messy and stale without us differentiating our different contract states the user has visited.

Reply:
> Do you feel like this solves the problem you've mentioned about the updating being indirect?

Yea that is exactly what i was thinking, looks good

> To be honest, I think we just need to add some global state to this application on second thinking. Because the user can hop around a lot and we're waiting for their TX to go through, we're just resisting the reality of the app.
> 
> The data will get very messy and stale without us differentiating our different contract states the user has visited.

I agree it seems like we do need it, its inevitable

Reply:
I do agree with it but I think this specific instance is different as its state that's only relevant to the table and very self contained